### PR TITLE
Docs / minimal CI update for Humble

### PR DESCRIPTION
**Pull request by luca-della-vedova**
### Fixed bug

Yet another doc / CI update for Humble! Minimal version to avoid introducing too much changes i.e. in the docker build workflow.

### CODE REVIEW

**Comment by luca-della-vedova** [comment]
Nightly fail is due to it trying to build the simulation / demos packages after the rename but the renaming PRs not being merged yet.

If there is any further issue with the nightly builds after the renaming it will be addressed in a future PR.

File: Dockerfile [hunk 1]
```diff
@@ -11,10 +11,11 @@ RUN apt-get update \
     git \
     python3-colcon-common-extensions \
     python3-vcstool \
-    qt5-default \
     wget \
     python3-pip \
+    clang-13 lldb-13 lld-13 \
```
Comment (Yadunund) [inline Dockerfile:16]: `Ubuntu 22.04` has `clang-14` so why `13`?

File: README.md [hunk 2]
```diff
@@ -62,23 +95,34 @@ Ensure all ROS 2 prerequisites are fulfilled,
...
+Compiling on `Ubuntu 22.04`:
+
+#### Install clang
+
+```bash
+sudo apt update
+sudo apt install clang lldb lld
+```
+
+**NOTE: RMF does not support building on gcc.**
```
Comment (Yadunund) [inline README.md:118]: I wouldn't say this.

```suggestion
**NOTE: We strongly recommend compiling Open-RMF packages with `clang` as compiler and `lld` as linker.**
```

File: README.md [hunk 3]
```diff
@@ -62,23 +95,34 @@ Ensure all ROS 2 prerequisites are fulfilled,
...
 
-Compiling on `Ubuntu 20.04`:
+#### Compile using clang
 
 ```bash
 cd ~/rmf_ws
-source /opt/ros/<your ros distro>/setup.bash
-colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release
+source /opt/ros/humble/setup.bash
+CXX=clang++ LDFLAGS='-fuse-ld=lld' colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release
```
Comment (Yadunund) [inline README.md:125]: We should promote using`colcon mixin`. It's a great tool that users will benefit from.

```suggestion
# Update colcon mixin which is one time step
colcon mixin add default https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml
colcon mixin update default

export CXX=clang++
export CX=clang
colcon build --mixin "release lld"
```

**Review by Yadunund — CHANGES_REQUESTED** [review]
Small comments

**Review by Yadunund — APPROVED** [review]

**Comment by orensbruli** [comment]
> her issue with the nightly builds after the renaming it will be addressed in a future P

For reference https://github.com/open-rmf/rmf_demos/pull/155

Is it only needed the sign-off of the last commit to getting it merged?

**Pull request merged**